File: Element Names and Symbols.py
n = int(input())
elements = {}
for i in range(n):
    ele,symbol = input().split()
    elements[ele]=symbol
choice = int(input())
if choice==1:
    symbol_search = input()
    for ele in elements:
        if elements[ele]==symbol_search:
            print(ele)
            exit(0)
else:
    ele_search = input()
    if ele_search in elements:
        print(elements[ele_search])
        exit(0)
    
print('Not found')

File: Element_Names_And_Symbols.cpp
#include <iostream>
using namespace std;
#include <string>
#include <map>
int main()
{
    int n,i; cin>>n;
    map<string,string> elements;
    for(i = 0; i < n; i++)
    {
        string ele,symbol;
        cin>>ele>>symbol;
        elements[ele] = symbol;
    }
    int choice; cin>>choice;
    
    
    if (choice==1)
    {
        string search_symbol; cin>>search_symbol; 
        map<string, string>::iterator it = elements.begin();
        int found = 0;
        while(it!=elements.end())
        {
            string ele = it->first;
            string symbol = it->second;
            if (symbol==search_symbol)
            {
                cout<<ele;
                found = 1;
            }
            it++;
            
        }
        if (found == 0)
            cout<<"Not found";
        
    }
       
        
    else
    {
        string search_ele; cin>>search_ele;
        if (elements.count(search_ele))
        {
            cout<<elements[search_ele];
            
        }
        else
            cout<<"Not found";
    }
    return 0;
}